### ISSUE TITLE
startOf('week') + weekStart edge case

### ISSUE THREAD

**Issue by ondblclick** [iamkun/dayjs#490]
looks like there is an edge case when
```
startOf('week')
```
in combination with `weekStart: n` gives wrong result:
```
import dayjs from 'dayjs';
import en from 'dayjs/locale/en-gb';

dayjs.locale({ ...en, weekStart: 1 });
console.log(dayjs('2019-02-10').startOf('week').format('YYYY-MM-DD'));
// => 2019-02-11
```
where `2019-02-10` is Sunday, `2019-02-11` is Monday - a start of the *next* week. Expected result in this case is `2019-02-04`, right?

**Comment by ondblclick** [comment]
momentjs has an `isoWeek` thingie, and the similar code for the example above gives correct result:
```
console.log(moment('2019-02-10').startOf('isoWeek').format('YYYY-MM-DD'));
// => 2019-02-04
```

**Comment by iamkun** [comment]
Thanks, the fix code has been merged into dev branch and will be released in next patch.

**Comment by ondblclick** [comment]
@iamkun awesome, thank you!

**Comment by iamkun** [comment]
fixed in v1.8.6
